### PR TITLE
Fix PWM configuration and usage

### PR DESCRIPTION
**Pull request by Qbicz**
After recent mergeup, PWM driver API changed so it requires pin number instead of PWM channel number. Also, configuration is kept in DTS instead of Kconfig.

### CODE REVIEW

**Comment by Qbicz** [comment]
@pdunaj @gamnes, can you review?

File: samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf [hunk 1]
```diff
@@ -42,22 +42,8 @@ CONFIG_PWM_NRF5_SW=n
 CONFIG_PWM_NRFX=y
 
 CONFIG_PWM_0=y
-CONFIG_PWM_0_NAME="PWM_0"
```
Comment (gamnes) [inline samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf]: Just a comment here: The names are still set through Kconfig - no change here. So if there was a reason for you to write this name in your .conf file, then that reason should still be there - but as far as I can tell, you were just using the default name of the driver so its OK to remove this.

Comment (Qbicz) [reply samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf]: Actually this is set with DT_NORDIC_NRF_PWM_PWM_0_LABEL, so I can use it.

Comment (Qbicz) [reply samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf]: The label comes from dtsi file

**Comment by Qbicz** [comment]
@carlescufi can you do next merge-up? Currently there is mistake in PWM DTS which was fixed today: https://github.com/zephyrproject-rtos/zephyr/pull/12137

**Comment by Qbicz** [comment]
@carlescufi actually this is non-blocking warning, so no need to hurry with the merge-up.

File: samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf (hunk 1 above)
Comment (Qbicz) [reply samples/nrf_desktop/configuration/app_nrf52840_pca20041.conf]: Discussed offline, leaving Kconfig name which is the one actually used.

File: samples/nrf_desktop/src/hw_interface/leds.c [hunk 2]
```diff
@@ -31,30 +32,45 @@ struct led {
 	struct k_delayed_work work;
 };
 
+/* Mapping the PWM channels to pin numbers */
```
[the remark this answers is not in the document]
Comment (Qbicz) [reply samples/nrf_desktop/src/hw_interface/leds.c]: Moved the array to `led_state_${BOARD}.c` for each platform

**Comment by Qbicz** [comment]
Fixed the PWM for PCA63519 and PCA10056 boards too. For PCA10056 which is maintained in Zephyr I needed to add an overlay file.

**Comment by Qbicz** [comment]
@myrdyr @carlescufi The CI is showing errors which are not reproduced locally with recent fork of Zephyr. This branch is rebased on the latest master. Looks like the CI builds against some other branch/outdated Zephyr.

http://jenkins-ncs.nordicsemi.no:8080/blue/organizations/jenkins/fw-nrfconnect-nrf/detail/PR-353/3/pipeline/

**Comment by pdunaj** [comment]
@Qbicz CI builds against alta branch
Alta changes must be migrated to master and CI switched. I talked about that with @wiminordic yesterday to check when this can happen. He asked around but this is unknown.

Hi @carlescufi , maybe you know when alta will be migrated?

**Comment by carlescufi** [comment]
@ioannisg @rlubos could you please upmerge the nrf91 branch?

**Comment by carlescufi** [comment]
Merging since this is battle-tested against fw-nrfconnect-zephyr master.

**Pull request merged**